modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberEditorTest.ts

```diff
@@ -0,0 +1,72 @@
+import { UiFinder } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { SugarBody } from '@ephox/sugar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.themes.silver.throbber.ThrobberEditorTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+  }, [ Theme ]);
+
+  const setProgressState = (editor: Editor, state: boolean, time?: number) => {
+    if (state) {
+      editor.setProgressState(true, time);
+    } else {
+      editor.setProgressState(false);
+    }
+  };
+
+  it('TINY-7373: should not change the editor selection', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>abc</p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 2);
+    setProgressState(editor, true);
+    await UiFinder.pWaitForVisible('Wait for throbber to show', SugarBody.body(), '.tox-throbber');
```

Comment:
Could these wait for's be pulled out into a separate function to remove the duplication in each test? You could possibly even pull out the whole lot. eg:
```
const pTestToggle = async (editor: Editor, action: () => void = Fun.noop) => {
  setProgressState(editor, true);
  await UiFinder.pWaitForVisible('Wait for throbber to show', SugarBody.body(), '.tox-throbber');
  action();
  setProgressState(editor, false);
  await UiFinder.pWaitForHidden('Wait for throbber to hide', SugarBody.body(), '.tox-throbber');
}
```

Reply:
Thanks for the suggestion. Looks a lot cleaner now.